File: examples/nodejs/07_complex_structures.js
const Redis = require("ioredis");

const redis = new Redis({ host: "localhost", port: 6379 });

async function main() {
    console.log("--- 07 Complex Structures ---");

    // Lists
    await redis.rpush("queue:jobs", "job1", "job2", "job3");
    const job = await redis.lpop("queue:jobs");
    console.log(`Processed: ${job}`);

    // Sets
    await redis.sadd("tags", "frontend", "javascript", "nodejs");
    const isMember = await redis.sismember("tags", "rust");
    console.log(`Is 'rust' a tag? ${isMember}`);

    // Hash
    await redis.hset("user:session:99", {
        "ip": "127.0.0.1",
        "last_login": Date.now()
    });
    const ip = await redis.hget("user:session:99", "ip");
    console.log(`Session IP: ${ip}`);

    redis.disconnect();
}

main();

File: examples/python/07_complex_structures.py
import redis

# Connect to Zedis
r = redis.Redis(host='localhost', port=6379, decode_responses=True)

print("--- 07 Complex Structures ---")

# LISTS
print("List Operations:")
r.rpush('mylist', 'A', 'B', 'C')
item = r.lpop('mylist')
print(f"Popped: {item}, Remaining: {r.lrange('mylist', 0, -1)}")

# HASHES
print("\nHash Operations:")
r.hset('myhash', mapping={'field1': 'value1', 'field2': 'value2'})
val = r.hget('myhash', 'field1')
print(f"Field1: {val}")

# SETS (Standard Redis Sets)
print("\nSet Operations:")
r.sadd('myset', 'apple', 'banana', 'apple') # 'apple' added once
members = r.smembers('myset')
print(f"Set members: {members}")
